doc/lint-lifecycle.md

```diff
@@ -0,0 +1,66 @@
+# The Life of a Lint
+
+## Lint States
+
+### Proposed
+
+A lint begins its life as a **proposal** which is _pending_ until _accepted_.
+
+### Accepted
+
+After discussion and sufficient agreement within the team, a proposed lint becomes **accepted**.
+
+An accepted lint is ready for implementation.
+
+### Experimental
+
+Lints can be marked experimental. 
+
+Reasons a lint might be marked experimental include a lint
+
+* being tentatively introduced
+* having unknown value
+* having incomplete semantics
+* having outstanding (fixable) false positives or
+* that might be known to be temporary.
+
+### Stable
+
+Stable lints are lints whose semantics and implementation are considered complete. False
+positives (that are not known, accepted and documentted limitations of lint semantics) are
+considered bugs and fixing them should be prioritized. False negatives may be bugs or
+enhancements (depending on semantics).
+
+### Deprecated
+
+Deprecated lints are lints that we plan to remove.
+
+Reasons for deprecation include:
+
+* semantics that don't make sense with current language semantics (e.g., null-safety)
+* stale advice
+* poor performance or
+* poor user experience (e.g., too many false positives).
+
+### Removed
+
+Lints we no longer support are removed.
+
+In general removal is preceded by a period of deprecation.
+
+## State Transitions
+
+### Deprecation
+
+Implemented lints can be deprecated.
+
+Deprecating lints that are in common lint sets (e.g., in [package:lints](https://github.com/dart-lang/lints)
+can be impactful so should be done with care.
+
+### Marking Stable
+
+Experimental lints should aspire to be stable. An experimental lint is a candidate for stable when it has
+
+* complete semantics
+* complete implementation (with no false-positives)
+* established long-term value (e.g., inclusion in a recommended lint set)
```

Comment:
Well... I don't know about how the recommended lint sets work. But it seems that the requirements would go the other way: if someone wanted an experimental lint rule to be included in a recommended lint set, the response would be that it needs to meet the qualifications of being stable before it can be recommended.

Reply:
Sorry, I meant _desire_ to be included.  That said, folks in general don't seem to pay much attention to these states and so they're if anything soft signals.  (But maybe we can change that?)

Reply:
I don't think that a desire to get a lint into the core or recommended lint sets should be a reason to make it stable. But I do think that a commitment on our part to support the lint is part of the criteria.

Also, similar question as above: should this section be moved under "Stable"?